packages/program/src/functions/invocation-results.ts

```diff
@@ -97,10 +98,14 @@ export class InvocationResult<T = any> {
       return callResultToInvocationResult<T>(callResult, callConfig, logs);
     }
 
+    const outputAsIterableHeap =
+      (callConfig?.func.encoding === ENCODING_V0 && callConfig?.func.outputMetadata.isHeapType) ||
+      false;
+
     const encodedResults = decodeContractCallScriptResult(
       callResult,
       (callConfig?.program as AbstractContract).id,
-      callConfig?.func.outputMetadata.isHeapType || false,
+      outputAsIterableHeap,
```

Comment:
The invocation result needs to know whether this is a `v0` or `v1` heap. Under `v0` it needs to iterate over receipts, under `v1` it can take all this data from the `ReturnData` reciept.

Reply:
Should we be considering completely purging the v0-related stuff?